### PR TITLE
circulation: upgrade to invenio-circulation v1.0.0a16

### PR DESCRIPTION
**Pull request by reropag**
* Changes loan_pid to pid.
* Improves checkout duration type from int to timedelta.

Co-Authored-by: Gianni Pante <gianni.pante@rero.ch>

# to test:
bootstrap, setup
test circulation transactions

### CODE REVIEW

File: rero_ils/modules/items/api.py [hunk 1]
```diff
@@ -136,12 +136,12 @@ def wrapper(item, *args, **kwargs):
...
         else:
             raise MissingRequiredParameterError(
-                description="Parameter 'loan_pid' is required")
+                description="Parameter 'pid' is required")
 
         # set missing parameters
         kwargs['item_pid'] = item.pid
         kwargs['patron_pid'] = loan.get('patron_pid')
-        kwargs['loan_pid'] = loan.get('loan_pid')
+        kwargs['pid'] = loan.get('pid')
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api.py]: loan.pid ?

File: rero_ils/modules/items/api.py [hunk 2]
```diff
@@ -622,12 +622,12 @@ def checkin(self, current_loan, **kwargs):
...
         if requests:
             request = next(self.get_requests())
-            requested_loan = Loan.get_record_by_pid(request.get('loan_pid'))
+            requested_loan = Loan.get_record_by_pid(request.get('pid'))
             pickup_location_pid = requested_loan.get('pickup_location_pid')
             if self.last_location_pid == pickup_location_pid:
                 if loan.is_active:
                     item, cancel_action = self.cancel_loan(
-                        loan_pid=loan.get('loan_pid'))
+                        pid=loan.get('pid'))
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api.py]: loan.pid ?

File: rero_ils/modules/items/api_views.py [hunk 3]
```diff
@@ -117,20 +117,20 @@ def librarian_request(item, data):
...
+        loan = Loan.get_record_by_pid(data.get('pid'))
         if (
             loan.get('state') == 'ITEM_IN_TRANSIT_FOR_PICKUP' and
             loan.get('patron_pid') == data.get('patron_pid')
         ):
             item.receive(**data)
         if loan.get('state') == 'ITEM_IN_TRANSIT_TO_HOUSE':
-            item.cancel_loan(loan_pid=loan.get('loan_pid'))
-            del data['loan_pid']
+            item.cancel_loan(pid=loan.get('pid'))
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api_views.py]: loan.pid ?

File: rero_ils/modules/items/api_views.py [hunk 4]
```diff
@@ -117,20 +117,20 @@ def librarian_request(item, data):
...
+            del data['pid']
     else:
         loan = get_loan_for_item(item.pid)
         if loan:
-            item.cancel_loan(loan_pid=loan.get('loan_pid'))
+            item.cancel_loan(pid=loan.get('pid'))
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api_views.py]: loan.pid ?

File: rero_ils/modules/items/api_views.py [hunk 5]
```diff
@@ -282,7 +282,7 @@ def item(item_barcode):
         abort(404)
     loan = get_loan_for_item(item.pid)
     if loan:
-        loan = Loan.get_record_by_pid(loan['loan_pid']).dumps_for_circulation()
+        loan = Loan.get_record_by_pid(loan.get('pid')).dumps_for_circulation()
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api_views.py]: loan.pid ?

File: rero_ils/modules/notifications/api.py [hunk 6]
```diff
@@ -226,7 +226,7 @@ def document(self):
 def get_availability_notification(loan):
     """Returns availability notification from loan."""
     results = NotificationsSearch().filter(
-        'term', loan__pid=loan.get('loan_pid')
+        'term', loan__pid=loan.get('pid')
```
Comment (Garfield-fr) [inline rero_ils/modules/notifications/api.py]: loan.pid ?

File: rero_ils/modules/notifications/api.py [hunk 7]
```diff
@@ -238,7 +238,7 @@ def get_availability_notification(loan):
 def get_recall_notification(loan):
     """Returns availability notification from loan."""
     results = NotificationsSearch().filter(
-        'term', loan__pid=loan.get('loan_pid')
+        'term', loan__pid=loan.get('pid')
```
Comment (Garfield-fr) [inline rero_ils/modules/notifications/api.py]: loan.pid ?

File: rero_ils/modules/notifications/api.py [hunk 8]
```diff
@@ -250,7 +250,7 @@ def get_recall_notification(loan):
 def number_of_reminders_sent(loan):
     """Get the number of overdue notifications sent for the given loan."""
     results = NotificationsSearch().filter(
-        'term', loan__pid=loan.get('loan_pid')
+        'term', loan__pid=loan.get('pid')
```
Comment (Garfield-fr) [inline rero_ils/modules/notifications/api.py]: loan.pid ?

File: tests/api/test_notifications_rest.py [hunk 9]
```diff
@@ -350,7 +350,7 @@ def test_recall_notification(client, patron_martigny_no_email,
     assert loan.is_notified(notification_type='recall')
 
     notification = get_recall_notification(loan)
-    assert notification.loan_pid == loan.get('loan_pid')
+    assert notification.loan_pid == loan.get('pid')
```
Comment (Garfield-fr) [inline tests/api/test_notifications_rest.py]: loan.pid ?

File: tests/fixtures/circulation.py [hunk 10]
```diff
@@ -701,7 +701,7 @@ def loan_validated_martigny(
     loan = list(item2_lib_martigny.get_loans_by_item_pid(
         item_pid=item2_lib_martigny.pid))[0]
     item2_lib_martigny.validate_request(
-        loan_pid=loan.get('loan_pid'),
+        pid=loan.get('pid'),
```
Comment (Garfield-fr) [inline tests/fixtures/circulation.py]: loan.pid ?

File: rero_ils/modules/items/api_views.py (hunk 3 above)
Comment (reropag) [reply rero_ils/modules/items/api_views.py]: not an invenio_circulation loan class, does not have the pid shortcut

File: rero_ils/modules/items/api_views.py (hunk 5 above)
Comment (reropag) [reply rero_ils/modules/items/api_views.py]: it is an invenio_circulation loan class which does not have the pid shortcut

File: rero_ils/modules/items/api_views.py (hunk 4 above)
Comment (reropag) [reply rero_ils/modules/items/api_views.py]: it is an invenio_circulation loan class which does not have the pid shortcut

**Comment by reropag** [comment]
> `./scripts/setup` fail at the line:
> `pipenv run invenio fixtures create --pid_type cipo ${DATA_PATH}/circulation_policies.json`

Done

**Pull request merged**